src/map/skill.c

```diff
@@ -6071,6 +6146,108 @@ static int skill_castend_damage_id(struct block_list *src, struct block_list *bl
+				// hit, right? For now well just use this logic until we can get a confirm on if it does this or not. [Rytech]
+				if (skill->attack(BF_WEAPON, src, src, bl, skill_id, skill_lv, tick, flag) > 0) { // Add the ID of the tagged target to the player's tag list and start the status on the target.
+					sd->stellar_mark[i] = bl->id;
+					// Val4 flags if the status was applied by a player or a monster.
+					// This will be important for other skills that work together with this one.
+					// 1 = Player, 2 = Monster.
+					// Note: Because the attacker's ID and the slot number is handled here, we have to
+					// apply the status here. We can't pass this data to skill_additional_effect.
+					sc_start4(src, bl, SC_FLASHKICK, 100, src->id, i, skill_lv, 1, skill->get_time(skill_id, skill_lv));
+				}
+			} else if (md != NULL) { // Monsters can't track with this skill. Just give the status.
+				if (skill->attack(BF_WEAPON, src, src, bl, skill_id, skill_lv, tick, flag) > 0)
+					sc_start4(src, bl, SC_FLASHKICK, 100, 0, 0, skill_lv, 2, skill->get_time(skill_id, skill_lv));
+			}
+		}
+			break;
+		case SJ_FALLINGSTAR_ATK:
+			if (sd != NULL) { // If a player used the skill it will search for targets marked by that player. 
+				if (tsc != NULL && tsc->data[SC_FLASHKICK] != NULL && tsc->data[SC_FLASHKICK]->val4 == 1) { // Mark placed by a player.
+					int i = 0;
+
+					ARR_FIND(0, MAX_STELLAR_MARKS, i, sd->stellar_mark[i] == bl->id);
+					if (i < MAX_STELLAR_MARKS) {
+						skill->attack(BF_WEAPON, src, src, bl, skill_id, skill_lv, tick, flag);
+						skill->castend_damage_id(src, bl, SJ_FALLINGSTAR_ATK2, skill_lv, tick, 0);
+					}
+				}
+			} else if (tsc != NULL && tsc->data[SC_FLASHKICK] != NULL && tsc->data[SC_FLASHKICK]->val4 == 2) { // Mark placed by a monster.
+				// If a monster used the skill it will search for targets marked by any monster since they can't track their own targets.
+				skill->attack(BF_WEAPON, src, src, bl, skill_id, skill_lv, tick, flag);
+				skill->castend_damage_id(src, bl, SJ_FALLINGSTAR_ATK2, skill_lv, tick, 0);
+			}
+			break;
+		case SJ_NOVAEXPLOSING:
+		{
+			skill->attack(BF_MISC, src, src, bl, skill_id, skill_lv, tick, flag);
+
+			// We can end Dimension here since the cooldown code is processed before this point.
+			if (sc != NULL && sc->data[SC_DIMENSION] != NULL)
+				status_change_end(src, SC_DIMENSION, INVALID_TIMER);
+			else // Dimension not active? Activate the 2 second skill block penalty.
+				sc_start(src, &sd->bl, SC_NOVAEXPLOSING, 100, skill_lv, skill->get_time(skill_id, skill_lv));
+		}
+			break;
+		case SP_SOULEXPLOSION:
+			if (!((tsc != NULL &&
+				(tsc->data[SC_SOULLINK] != NULL ||
+				tsc->data[SC_SOULGOLEM] != NULL ||
+				tsc->data[SC_SOULSHADOW] != NULL ||
+				tsc->data[SC_SOULFALCON] != NULL ||
+				tsc->data[SC_SOULFAIRY] != NULL))
+				|| tstatus->hp < 10 * tstatus->max_hp / 100
```

Comment:
add some brackets here to make it easier to read

Reply:
hmmm interesting format to be done here.

Reply:
LOL after testing i noticed the code is wrong since all are inside `!` LOL